Gems/Multiplayer/Code/Source/Components/NetworkCharacterComponent.cpp

```diff
@@ -107,10 +107,22 @@ namespace Multiplayer
     {
     }
 
+
     void NetworkCharacterComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
     {
-        Physics::CharacterRequests* characterRequests = Physics::CharacterRequestBus::FindFirstHandler(GetEntityId());
-        m_physicsCharacter = (characterRequests != nullptr) ? characterRequests->GetCharacter() : nullptr;
+        // During activation the character controller is not created yet.
+        // Connect to CharacterNotificationBus to listen when it's activated after creation.
+        Physics::CharacterNotificationBus::Handler::BusConnect(GetEntityId());
+    }
+
+    void NetworkCharacterComponent::OnCharacterActivated(const AZ::EntityId& entityId)
+    {
+        Physics::CharacterNotificationBus::Handler::BusDisconnect();
```

Comment:
Notice how `NetworkCharacterComponent` is only using `CharacterNotificationBus` for "creation" only since it disconnects right away.

I did this to keep the exact same logic as before since it wasn't handling the character being enabled/disabled. But having into consideration that enabling/disabling a character recreates the body entirely internally my suggestion would be to keep listening to `OnCharacterActivated` and `OnCharacterDeactivated` to refresh everything inside `NetworkCharacterComponent`. The only reason I didn't doit is because I didn't know the consequences of such change and so I kept the same logic as before.

This is now exactly the same as in NetworkRigidBodyComponent and NetworkRigidBodyControllerComponent, so they are aligned now, but if you guys prefer to go ahead and change it please change it in all of them.

Reply:
@lmbr-pip @kberg-amzn would be doing `m_physicsCharacter = nullptr;` in a `OnCharacterDeactivated` function enough to revert all that `OnCharacterActivated` is doing or would it need to do something else. Is it harmless that if the character is activate again all the code of `OnCharacterActivated` is executed again?

Reply:
@lmbr-pip @kberg-amzn I'm not very familiar with this code, but I'm from looking at the OnCharacterActivated(), I'm thinking we need the following?
- Add the OnCharacterDeactivated() to set m_physicsCharacter to null.
- Move these lines to Activate() instead of OnCharacterActivated(), so that way in case OnCharacterActivated/Deactivated ever gets called more than once, things continue to work ok. These seem like they shouldn't get called more than once, and that they're ok to call even before the character exists?
```
        GetNetBindComponent()->AddEntitySyncRewindEventHandler(m_syncRewindHandler);
```
```
        if (!HasController())
        {
            GetNetworkTransformComponent()->TranslationAddEvent(m_translationEventHandler);
        }
```

Reply:
I would not move the code to `Activate()` because those register to call the function `NetworkCharacterComponent::OnSyncRewind` which needs `m_physicsCharacter` anyway, so I think it's more correct to do it all `OnCharacterActivated`.

Probably `OnCharacterDeactivated` would need to register those events too. Right?

Reply:
By the way, I'd be inclined to leave this for a subsequent PR.

Reply:
i discussed with @AMZN-Olex and he's in favor of moving those chunks to Activate(), and adding the Disconnect() calls to Deactivate(), so that the network handling is kept more obviously separated from the physics character handling. i'll put that up for review in a follow-up PR.